lab_3_ann_retriever/main.py

```diff
@@ -541,6 +638,19 @@ def retrieve_relevant_documents(
 
         In case of corrupt input arguments, None is returned.
         """
+        if (not query or not isinstance(query, str) or not isinstance(n_neighbours, int)
+                or isinstance(n_neighbours, bool)):
+            return None
+        query_vector = self._index_document(query)
+        if not query_vector:
+            return None
+        knn_list = self._calculate_knn(query_vector, self._document_vectors, n_neighbours)
+        if not knn_list:
+            return None
+        result = []
```

Comment:
list comprehension

Reply:
where?

Reply:
sorry, i misunderstood it.. is it better now?